pkg/traversal/traversal_test.go

```diff
@@ -1,122 +1,49 @@
 package traversal
 
 import (
-	"bytes"
-	"io"
-	"io/ioutil"
-	"os"
-	"path/filepath"
 	"testing"
 
-	"github.com/stretchr/testify/require"
-	"gopkg.in/yaml.v2"
-
-	"github.com/openshift/must-gather-clean/pkg/input"
-	"github.com/openshift/must-gather-clean/pkg/obfuscator"
-	"github.com/openshift/must-gather-clean/pkg/omitter"
-	"github.com/openshift/must-gather-clean/pkg/output"
+	"github.com/stretchr/testify/assert"
 )
 
-type noopObfuscator struct {
-	replacements map[string]string
-}
-
-func (d noopObfuscator) GetReplacement(original string) string {
-	return original
-}
-
-func (d noopObfuscator) Path(input string) string {
-	return input
-}
-
-func (d noopObfuscator) Contents(input string) string {
-	return input
+type collectingQueueProcessor struct {
+	paths []string
 }
 
-func (d noopObfuscator) Report() map[string]string {
-	return d.replacements
-}
-
-func (d noopObfuscator) ReportReplacement(original string, replacement string) {
-
-}
-
-type inMemfile struct {
-	Contents    string      `yaml:"contents"`
-	Permissions os.FileMode `yaml:"permissions"`
-}
-
-type testContents struct {
-	Files map[string]string `yaml:"files"`
-}
-
-type memoryOutputter struct {
-	t     *testing.T
-	Files map[string]inMemfile
-}
-
-func (m *memoryOutputter) Writer(relPath string, permissions os.FileMode) (output.Closer, io.StringWriter, error) {
-	require.NotContains(m.t, m.Files, relPath)
-	buffer := &bytes.Buffer{}
-	return func() error {
-		m.Files[relPath] = inMemfile{Contents: buffer.String(), Permissions: permissions.Perm()}
-		return nil
-	}, buffer, nil
-}
-
-func testOutputter(t *testing.T) *memoryOutputter {
-	return &memoryOutputter{t: t, Files: map[string]inMemfile{}}
+func (c *collectingQueueProcessor) ProcessQueue(queue chan WorkerInput, _ chan error) {
+	for wf := range queue {
+		c.paths = append(c.paths, wf.path)
+	}
 }
 
 func TestFileWalker(t *testing.T) {
 	for _, tc := range []struct {
-		name     string
-		inputDir string
+		name           string
+		inputDir       string
+		expectedResult []string
 	}{
 		{
 			name:     "basic",
-			inputDir: "testfiles/test1",
+			inputDir: "testfiles/test1/mg",
+			expectedResult: []string{
```

Comment:
It appears the tests are not using the reports file anymore. They should be removed.

Reply:
which reports? they are now in a dedicated struct

Reply:
This one https://github.com/openshift/must-gather-clean/blob/main/pkg/traversal/testfiles/test1/report.yaml